webrender/src/device/gfx/device.rs

```diff
@@ -1270,14 +1278,59 @@ impl<B: hal::Backend> Device<B> {
                 &program
                     .pipelines
                     .get(&(format, blend_state, self.render_pass_depth_state, self.current_depth_test))
-                    .expect(&format!(
-                        "The blend state {:?} with depth test {:?} and render_pass_depth_state {:?} and format {:?} not found for {} program!",
-                        blend_state, self.current_depth_test, self.render_pass_depth_state, format, program.shader_name
-                    )),
+                    .expect(
+                        // Creating this string takes a lot of time, so we only enable it in debug
+                        #[cfg(debug_assertions)]
+                        &format!(
```

Comment:
We can instead do `or_else(|| panic!(...))` which would only run for the error case